benchmarking/src/time_bench.rs

```diff
@@ -0,0 +1,313 @@
+                }
+            }
+        }
+        Err(err) => {
+            eprintln!("could not run cargo: {err}");
+        }
+    }
+}
+
+/// Build the `time` binary, which is an implicit dependency of benchmarks.
+pub fn compile_time_bin() {
+    run_cargo(["build", "--release", "--bin", "time"]);
+}
+
+/// Compile the `temporal-verifier` binary, which is run by the benchmarks.
+pub fn compile_flyvy_bin() {
+    run_cargo(["build", "--release", "--bin", "temporal-verifier"]);
+}
+
+#[derive(Debug, Clone)]
+struct RawRunMeasurement {
+    real_time: Duration,
+    usage: Usage,
+    self_usage: Usage,
+    timed_out: bool,
+    status: Option<i32>,
+    signal: Option<Signal>,
+}
+
+fn time_val_to_duration(time: TimeVal) -> Duration {
+    Duration::from_secs(time.tv_sec() as u64) + Duration::from_micros(time.tv_usec() as u64)
+}
+
+impl RawRunMeasurement {
+    fn into_result(self) -> RunMeasurement {
+        RunMeasurement {
+            real_time: self.real_time,
+            user_time: time_val_to_duration(self.usage.user_time()),
+            sys_time: time_val_to_duration(self.usage.system_time()),
+            self_user_time: time_val_to_duration(self.self_usage.user_time()),
+            max_mem_bytes: self.usage.max_rss() as usize,
+            timed_out: self.timed_out,
+            success: self.status == Some(0),
+        }
+    }
+}
+
+impl Time {
+    fn get_child_results(&self, child: i32) -> Result<RawRunMeasurement, io::Error> {
+        let child = Pid::from_raw(child);
+        let start = Instant::now();
+        let timeout = Arc::new(Mutex::new(false));
+        if let Some(limit) = &self.time_limit {
+            let limit: Duration = (*limit).into();
+            let pgid = getpgid(Some(child)).expect("could not get child pgid");
+            let timeout = timeout.clone();
+            thread::spawn(move || {
+                thread::sleep(limit);
+                *timeout.lock().unwrap() = true;
+                let r = killpg(pgid, Signal::SIGTERM);
+                if let Err(err) = r {
+                    eprintln!("could not send SIGTERM: {err}");
+                }
+                thread::sleep(Duration::from_secs(1));
+                eprintln!("sending SIGKILL");
+                _ = killpg(pgid, Signal::SIGKILL);
+            });
+        }
+
+        let wstatus = waitpid(Some(child), None)?;
+        let status = match wstatus {
+            WaitStatus::Exited(_, status) => Some(status),
+            _ => None,
+        };
+        let signal = match wstatus {
+            WaitStatus::Signaled(_, signal, _) => Some(signal),
+            _ => None,
+        };
+        let real_time = start.elapsed();
+        let usage = getrusage(UsageWho::RUSAGE_CHILDREN)?;
+        let self_usage = getrusage(UsageWho::RUSAGE_SELF)?;
+        let measurement = RawRunMeasurement {
+            real_time,
+            usage,
+            self_usage,
+            timed_out: *timeout.lock().unwrap(),
```

Comment:
Makes sense, will fix.

The race between finishing normally and the timeout could be improved to correctly handle cases where the timeout thread is actually too late, but I think it's not worth handling this precisely since the program was so close to the deadline.